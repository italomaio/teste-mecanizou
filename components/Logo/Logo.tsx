import React from "react";

import Image from "next/image";
import LogoImage from "@/assets/images/mecanizou-logo-dark.avif";
import { cn } from "@/utils/classes";

export type LogoProps = {
  animate?: boolean;
  bordered?: boolean;
  className?: string;
};

const Logo: React.FC<LogoProps> = ({
  animate = true,
  bordered = true,
  className,
}) => {
  return (
    <Image
      src={LogoImage}
      alt="Logo mecanizou"
      width={150}
      height={150}
      className={cn("py-4 px-6", className, {
        "scale-transition": animate,
        "border border-white": bordered,
      })}
    />
  );
};

export default Logo;
