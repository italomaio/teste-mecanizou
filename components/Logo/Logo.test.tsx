import React from "react";
import { render } from "@testing-library/react";
import Logo, { LogoProps } from "./Logo";

describe("Logo Component", () => {
  const renderLogo = (props: Partial<LogoProps> = {}) => {
    const defaultProps: LogoProps = {
      animate: true,
      bordered: true,
      className: "",
    };
    return render(<Logo {...defaultProps} {...props} />);
  };

  it("Should render the logo image", () => {
    const { getByAltText } = renderLogo();
    const logoImage = getByAltText("Logo mecanizou");
    expect(logoImage).toBeInTheDocument();
  });

  it("Should apply animation class when animate is true", () => {
    const { getByAltText } = renderLogo({ animate: true });
    const logoImage = getByAltText("Logo mecanizou");
    expect(logoImage).toHaveClass("scale-transition");
  });

  it("Should not apply animation class when animate is false", () => {
    const { getByAltText } = renderLogo({ animate: false });
    const logoImage = getByAltText("Logo mecanizou");
    expect(logoImage).not.toHaveClass("scale-transition");
  });

  it("Should apply border class when bordered is true", () => {
    const { getByAltText } = renderLogo({ bordered: true });
    const logoImage = getByAltText("Logo mecanizou");
    expect(logoImage).toHaveClass("border border-white");
  });

  it("Should not apply border class when bordered is false", () => {
    const { getByAltText } = renderLogo({ bordered: false });
    const logoImage = getByAltText("Logo mecanizou");
    expect(logoImage).not.toHaveClass("border border-white");
  });

  it("Should apply custom className", () => {
    const customClass = "custom-class";
    const { getByAltText } = renderLogo({ className: customClass });
    const logoImage = getByAltText("Logo mecanizou");
    expect(logoImage).toHaveClass(customClass);
  });

  it("Should match the snapshot", () => {
    const { asFragment } = renderLogo();
    expect(asFragment()).toMatchSnapshot();
  });
});
