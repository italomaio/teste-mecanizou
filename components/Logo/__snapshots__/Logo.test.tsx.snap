// Jest Snapshot v1, https://jestjs.io/docs/snapshot-testing

exports[`Logo Component Should match the snapshot 1`] = `
<DocumentFragment>
  <img
    alt="Logo mecanizou"
    class="py-4 px-6 dark:invert scale-transition border border-white"
    height="150"
    src="test-file-stub"
    width="150"
  />
</DocumentFragment>
`;
